tools/docker/generate_dockerfiles.py

```diff
@@ -863,7 +863,7 @@ def install_deps_spack() -> str:
 # Install a recent Spack version
 WORKDIR /root/spack
 ARG SPACK_VERSION
-ENV SPACK_VERSION=${{SPACK_VERSION:-a3abc1c492f2431f477a63bbccb48aa3a2d34199}}
+ENV SPACK_VERSION=${{SPACK_VERSION:-develop}}
```

Comment:
I believe this will make the build unreproducible. Ideally we'd use a tag like `develop-2025-03-30`, but a commit sha would also work.

Reply:
Yes, indeed, we can move back to a fixed tag for the `SPACK_VERSION` after we can keep now very recent (proprietary) spack recipes within the CP2K tree if needed.